src/Roslyn.Diagnostics.Analyzers/UnitTests/CSharpAvoidOptSuffixForNullableEnableCodeTests.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Roslyn.Diagnostics.CSharp.Analyzers.CSharpAvoidOptSuffixForNullableEnableCode,
+    Roslyn.Diagnostics.CSharp.Analyzers.CSharpAvoidOptSuffixForNullableEnableCodeCodeFixProvider>;
+
+namespace Roslyn.Diagnostics.Analyzers.UnitTests
+{
+    public class CSharpAvoidOptSuffixForNullableEnableCodeTests
+    {
+        [Fact]
+        public async Task RS0046_CSharp8_NullableEnabledCode_Diagnostic()
```

Comment:
Can you also add a test to verify no diagnostic for below case:

```csharp
#nullable enable
public class Class1
{
    private Class1 _instanceOpt, otherInstanceOpt;
    public void Method1(string sOpt)
    {
        string localOpt, otherLocalOpt;
    }
}
```

Reply:
Can you also add positive and negative tests for nullable value types and value types in nullable enable context?

Reply:
A diagnostic is reported on your example. I don't know how to tell if the symbol is nullable or not, any idea?